### PR TITLE
PPNA-6028 Add product custom dimensions

### PR DESCRIPTION
**Pull request by pulina**
Inner Documentation for product custom dimensions for comperasion: https://new-hope-travis.s3.amazonaws.com/PiwikPRO/PPMS-Analytics-Reporting/@dev/docs/index.html?path=analytics.yaml
Local view
image

### CODE REVIEW

File: custom_reports/object_management_api/index.yaml [hunk 1]
```diff
@@ -19,6 +19,10 @@ paths:
     $ref: ./paths/custom_dimensions_detail.yaml
   /api/analytics/v1/manage/custom-dimensions/slots/:
     $ref: ./paths/custom_dimensions_slots.yaml
+  /api/analytics/v1/manage/product-custom-dimensions/:
```
Comment (mrfijas) [inline custom_reports/object_management_api/index.yaml:22]: hmm.. we don't have these docs in [reporting](https://github.com/PiwikPRO/PPMS-Analytics-Reporting/blob/8ef256280624177c8589c03a90bd1ce74994d7fb/docs/internal-object-management.yaml), right?

I think we should either go with auto-generated docs or validate their in CI during reporting build against real API - currently it unmaintainable :/
yet another great dev-friday task xD

File: custom_reports/object_management_api/definitions/product_custom_dimension.yaml [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+    properties:
+      website_id:
+        type: string
+        format: uuid
+        description: ID of the website.
+        example: dde2dc68-b6bb-45ee-9ac2-868b6181e33f
+      name:
+        type: string
+        description: Name of the product custom dimension.
+        example: Visitor hair color
```
Comment (mrfijas) [inline custom_reports/object_management_api/definitions/product_custom_dimension.yaml:31]:
```suggestion
        example: Color
```
? I don't think shops often have products with `Visitor hair color` property

File: custom_reports/object_management_api/definitions/product_custom_dimension_details.yaml [hunk 3]
```diff
@@ -0,0 +1,14 @@
+type: object
+allOf:
+  - $ref: '../index.yaml#/components/schemas/ProductCustomDimension'
```
Comment (mrfijas) [inline custom_reports/object_management_api/definitions/product_custom_dimension_details.yaml:3]: oh my, our PublicDocs are horrible to read, with all this jumping between files :/

**Review by mrfijas — APPROVED** [review]

**Review by kororokke — APPROVED** [review]
18.3

File: custom_reports/object_management_api/definitions/product_custom_dimension.yaml (hunk 2 above)
Comment (kororokke) [reply custom_reports/object_management_api/definitions/product_custom_dimension.yaml:31]: only for readheads 😆 :trollface:

Comment (pulina) [reply custom_reports/object_management_api/definitions/product_custom_dimension.yaml]: I changed this already except i do not push changes ... :/

File: custom_reports/object_management_api/index.yaml (hunk 1 above)
Comment (pulina) [reply custom_reports/object_management_api/index.yaml:22]: Oh god what a mess. We should move our docs to company one and durig this process we need to clean this up somehow.  This is to important to be pet-project.

**Pull request merged**